README.md

```diff
@@ -1,9 +1,47 @@
+## PCL Mean and Covariance
+
 Experiments with mean and covariance computation in PCL.
 
-1. Clone this repository with submodules (pass `--recursive` flag)
+## Installation
+
+1. **Clone this repository** 
+
+   Clone with submodules and pass the `--recursive` flag:
+
+       $ git clone https://github.com/taketwo/pcl-mean-and-covariance.git --recursive-submodules
+
+2. **Configure with CMake**
+    
+       $ cd <path to>/pcl-mean-and-covariance
+       $ cmake -C CMakeLists.txt -G "Unix Makefiles"
+
+   Note: CMake 3.5 or higher is required. Also note that the generator specified after the `-G` can be something other than Unix Makefiles, E.g. Ninja. 
+
+3. **Install `datamash`** 
+
+   This is required to build and run the benchmark. On later versions of Ubuntu, this can be installed as a prebuilt package with the following command:
+
+       $ sudo apt install datamash
+
+   However on older systems it may be necessary to install `datamash` from source. See the [datamash download page](https://www.gnu.org/software/datamash/download/) for details.
+
+4. **Build and run benchmarks** 
+
+       $ cd ~/bench
```

Comment:
If you follow out-of-source build, then this is not necessary. It will be possible to `make benchmarks` or `make tests` directly from the `build` folder.